packages/client/lib/sync/fetcher/blockfetcherbase.ts

```diff
@@ -68,12 +68,13 @@ export abstract class BlockFetcherBase<JobResult, StorageItem> extends Fetcher<
 
     while (count >= BigInt(max) && tasks.length < maxTasks) {
       tasks.push({ first: first, count: max })
-      first += BigInt(max)
+      !this.reverse ? (first += BigInt(max)) : (first -= BigInt(max))
```

Comment:
LGTM though I guess I'm curious, do the existing reverse fetcher tests not catch this?  What's the difference between this scenario and whatever is happening today?  And if it is different, is it possible to add a test to cover this particular scenario?

Reply:
I will check and add some tests, though some other things seemingly need to be fixed as well.